pyproject.toml

```diff
@@ -1,41 +1,51 @@
-[project]
+[tool.poetry]
 name = "ngsPETSc"
-version = "0.0.5"
-description = "NGSolve/Netgen interface to PETSc."
-readme = "README.md"
+version = "0.0.7a7"
+description = "NGSolve/Netgen interface to PETSc"
 authors = [
-  {name = "Umberto Zerbinati", email = "umberto.zerbinati@maths.ox.ac.uk"},
-  {name = "Patrick E. Farrell", email = "patrick.farrell@maths.ox.ac.uk"},
-  {name = "Stefano Zampini", email = "stefano.zampini@kaust.edu.sa"},
-  {name = "Jack Betteridge", email = "J.Betteridge@imperial.ac.uk"},
+  "Umberto Zerbinati <umberto.zerbinati@maths.ox.ac.uk>",
+  "Patrick E. Farrell <patrick.farrell@maths.ox.ac.uk>",
+  "Stefano Zampini <stefano.zampini@kaust.edu.sa>",
+  "Jack Betteridge <J.Betteridge@imperial.ac.uk>",
 ]
 maintainers = [
-  {name = "Umberto Zerbinati", email = "umberto.zerbinati@maths.ox.ac.uk"},
+  "Umberto Zerbinati <umberto.zerbinati@maths.ox.ac.uk>",
 ]
-license = {file = "LICENSE.txt"}
-dependencies = [
-  "mpi4py",
-  "numpy",
-  "scipy",
+readme = "README.md"
+license = "MIT"
+packages = [
+    {include = "ngsPETSc"}
 ]
-requires-python = ">=3.8"
+documentation = "https://ngspetsc.readthedocs.io/en/latest/"
+repository = "https://github.com/NGSolve/ngsPETSc"
 classifiers = [
   "Programming Language :: Python :: 3",
   "License :: OSI Approved :: MIT License",
   "Operating System :: OS Independent",
   "Development Status :: 3 - Alpha",
 ]
 
-[project.urls]
-Documentation = "https://ngspetsc.readthedocs.io/en/latest/"
-Repository = "https://github.com/NGSolve/ngsPETSc"
+[tool.poetry.dependencies]
+python = "^3.9"
+netgen-mesher = "^6.2"
+netgen-occt = "^7.8"
+petsc4py =  "^3.22.1"
+mpi4py = "^4"
```

Comment:
If a look at ngsPETSc code, it seems there's no need to have `mpi4py` listed here, if not for a few occurrences of MPI.COMM_WORLD that can be replaced by `PETSc.COMM_WORLD.tompi4py()` .  @UZerbinati How mpi4py is handled by the netgen related packages? 

Just try to install with this branch and got this
```
$ mpiexec -n 2 python test_vec.py 
/home/szampini/Devel/sandbox/ngspetsc/ngsPETSc/ngsPETSc/eps.py:10: UserWarning: Import Warning: it was not possible to import SLEPc
  warnings.warn("Import Warning: it was not possible to import SLEPc")
/home/szampini/Devel/sandbox/ngspetsc/ngsPETSc/ngsPETSc/eps.py:10: UserWarning: Import Warning: it was not possible to import SLEPc
  warnings.warn("Import Warning: it was not possible to import SLEPc")
Assertion failed in file ./src/include/mpir_request.h at line 508: ((req))->ref_count >= 0
/usr/lib64/mpich/lib/libmpi.so.12(+0x2d88ff) [0x7f5c956d88ff]
/usr/lib64/mpich/lib/libmpi.so.12(+0x2d8b60) [0x7f5c956d8b60]
/usr/lib64/mpich/lib/libmpi.so.12(+0x202bd2) [0x7f5c95602bd2]
/usr/lib64/mpich/lib/libmpi.so.12(PMPI_Waitall+0x106) [0x7f5c95510e76]
/home/szampini/Devel/sandbox/ngspetsc/ngsPETSc/test-jack/lib64/python3.11/site-packages/netgen/../netgen_mesher.libs/libnglib.so(+0x391129) [0x7f5ccf191129]
/home/szampini/Devel/sandbox/ngspetsc/ngsPETSc/test-jack/lib64/python3.11/site-packages/netgen/../netgen_mesher.libs/libnglib.so(+0x38e252) [0x7f5ccf18e252]
/home/szampini/Devel/sandbox/ngspetsc/ngsPETSc/test-jack/lib64/python3.11/site-packages/netgen/../netgen_mesher.libs/libnglib.so(+0x38fff8) [0x7f5ccf18fff8]
/home/szampini/Devel/sandbox/ngspetsc/ngsPETSc/test-jack/lib64/python3.11/site-packages/netgen/../netgen_mesher.libs/libnglib.so(_ZN6netgen4Mesh10DistributeEv+0x87) [0x7f5ccf190097]
/home/szampini/Devel/sandbox/ngspetsc/ngsPETSc/test-jack/lib64/python3.11/site-packages/netgen/../netgen_mesher.libs/libnglib.so(+0x3df8a1) [0x7f5ccf1df8a1]
/home/szampini/Devel/sandbox/ngspetsc/ngsPETSc/test-jack/lib64/python3.11/site-packages/netgen/../netgen_mesher.libs/libnglib.so(+0x18466a) [0x7f5ccef8466a]
/lib64/libpython3.11.so.1.0(+0x1d56e1) [0x7f5ce4bd56e1]
/lib64/libpython3.11.so.1.0(_PyObject_MakeTpCall+0x73) [0x7f5ce4bb7c73]
/lib64/libpython3.11.so.1.0(_PyEval_EvalFrameDefault+0x7e9) [0x7f5ce4bc0779]
/lib64/libpython3.11.so.1.0(+0x1bc6fa) [0x7f5ce4bbc6fa]
/lib64/libpython3.11.so.1.0(PyEval_EvalCode+0xac) [0x7f5ce4c463ac]
/lib64/libpython3.11.so.1.0(+0x264473) [0x7f5ce4c64473]
/lib64/libpython3.11.so.1.0(+0x2609fa) [0x7f5ce4c609fa]
/lib64/libpython3.11.so.1.0(+0x276752) [0x7f5ce4c76752]
/lib64/libpython3.11.so.1.0(_PyRun_SimpleFileObject+0x1a9) [0x7f5ce4c75f49]
/lib64/libpython3.11.so.1.0(_PyRun_AnyFileObject+0x48) [0x7f5ce4c75be8]
/lib64/libpython3.11.so.1.0(Py_RunMain+0x2eb) [0x7f5ce4c6f79b]
/lib64/libpython3.11.so.1.0(Py_BytesMain+0x3b) [0x7f5ce4c3652b]
/lib64/libc.so.6(+0x27550) [0x7f5ce484a550]
/lib64/libc.so.6(__libc_start_main+0x89) [0x7f5ce484a609]
python(_start+0x25) [0x555a452c1095]
internal ABORT - process 0
```

```
$ pip list
Package            Version  Editable project location
------------------ -------- ----------------------------------------------
intel-cmplr-lib-ur 2025.0.0
intel-openmp       2025.0.0
mkl                2025.0.0
mpi4py             4.0.1
netgen-mesher      6.2.2405
netgen_occt        7.8.1
ngsolve            6.2.2405
ngsPETSc           0.0.7a7  /home/szampini/Devel/sandbox/ngspetsc/ngsPETSc
numpy              2.1.3
petsc4py           3.22.1
pip                22.2.2
scipy              1.14.1
setuptools         62.6.0
tbb                2022.0.0
tcmlib             1.2.0
umf                0.9.0
```

Reply:
This is not good ! I'll replace the use of `MPI.COMM_WORLD` as you suggested.

Reply:
I'm not sure how mpi4py is handled by the netgen related packages? @mhochsteger might know more since I remember he put a lot of work into making ngsolve with mpi support pip installable.

Reply:
Seems like it could be an issue with your MPI installation?
[4.0 is using large count datatypes](https://github.com/mpi4py/mpi4py/releases/tag/4.0.0), which causes some [headaches with existing MPI installations](https://gitlab.com/petsc/petsc/-/issues/1625) that don't conform to the MPI 4.0 standard.
It'd be nice to just suck down the mpi4py wheel, but that isn't available on PyPI. (...yet?)

Reply:
I don't understand why users _wouldn't_ want mpi4py TBH!

Reply:
Same issue with mpi4py 3.1.6, not a mpi4 problem. Does it work for you?  I have mpich on fedora and installed ngspetsc using   `pip install  -e .` letting the build figure out the dependencies

Reply:
> I don't understand why users _wouldn't_ want mpi4py TBH!

you can make the same argument with many other useful python packages. The point is that if a package does not. explicitly uses another one, it should not list it as a dependency. The only reason it is a dependency now is because of the misuse of `MPI.COMM_WORLD `

Reply:
https://github.com/NGSolve/netgen/issues/196

Reply:
In a fresh python virtual environment `pip install ngsPETSc ngsolve` allows me to run all tests.
I'm working on Archlinux and OpenMPI, I'll try next on a fresh install with  MPICH!